File: data/readable_format/cpp/file320.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

// Function declaration
void writeVector(const std::vector<double>& ad, int neq, int number);

int main() {
    // Test cases
    std::vector<std::vector<double>> vectors = {
        {1.0, 2.0, 3.0},
        {4.0, 5.0, 6.0},
        {7.0, 8.0, 9.0}
    };
    std::vector<int> nums = {1, 2, 3};

    for(size_t i = 0; i < nums.size(); ++i) {
        writeVector(vectors[i], vectors[i].size(), nums[i]);
        std::string fileName = "vector_" + std::string(1, char(nums[i] + 96)) + ".out";
        std::ifstream file(fileName);
        std::string line;
        if (file.is_open()) {
            while (getline(file, line)) {
                std::cout << line << std::endl;
            }
            file.close();
        } else {
            std::cerr << "Unable to open file: " << fileName << std::endl;
        }
    }

    return 0;
}

// Function definition
void writeVector(const std::vector<double>& ad, int neq, int number) {
    std::string name = "vector_" + std::string(1, char(number + 96)) + ".out";

    std::ofstream outFile(name); // Automatically opens the file
    if (!outFile) {
        std::cerr << "Failed to open file: " << name << std::endl;
        return;
    }

    outFile << "vector number " << number << std::endl;
    for (int i = 0; i < neq; ++i) {
        outFile << "row " << (i+1) << " value " << ad[i] << std::endl;
    }

    outFile.close(); // Automatically called by the destructor if not explicitly called
}

File: data/readable_format/fortran/file320.f90
program testWriteVector
    implicit none
    integer :: nums(3) = (/1, 2, 3/)
    real*8 :: vectors(3,3) = reshape((/1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0/), (/3,3/))
    integer :: i, j
    character(len=50) :: line

    ! Make sure to include or link your original writevector subroutine here.

    do i = 1, size(nums)
        call writevector(vectors(:,i), 3, nums(i))
        open(unit=20, file='vector_'//trim(adjustl(char(nums(i)+96)))//'.out', status='old')
        do j = 1, 4 + 3 ! Assuming header + number of rows
            read(20, '(A)') line
            print *, trim(adjustl(line))
        end do
        close(20)
    end do
end program testWriteVector

subroutine writevector(ad, neq, number)
    implicit none
    character(len=12) :: name
    integer :: neq, i, number
    real(8) :: ad(*)

    name = 'vector_' // char(iachar('a') + number - 1) // '.out'

    open(10, file=name, status='unknown')
    write(10, *) 'vector number ', number
    do i = 1, neq
        write(10, *) 'row ', i, ' value ', ad(i)
    end do
    close(10)
end subroutine writevector